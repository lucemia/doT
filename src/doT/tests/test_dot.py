from pathlib import Path

import pytest
from syrupy.extensions.json import JSONSnapshotExtension

from .. import template, template_settings


@pytest.fixture
def snapshot_json(snapshot):
    return snapshot.use_extension(JSONSnapshotExtension)


@pytest.mark.parametrize(
    "case", (Path(__file__).parent / "test_dot").glob("*.tmpl"), ids=lambda x: x.name
)
def test_template(snapshot_json, case: Path):
    setting = template_settings._replace(strip=False)

    with case.open("r", encoding="utf-8") as ifile:
        _template = ifile.read()
        assert snapshot_json == template(_template, setting)
