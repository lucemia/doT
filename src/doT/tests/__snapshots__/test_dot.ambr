# serializer version: 1
# name: test_template
  "function anonymous(it) {var out='';if(it.options == 1\\n){out+='Option 1';}else if(it.options == 2\\n){out+='Option 2';}else if(\\n){out+='Other option';}out+=':\\n\\n More text';return out;}"
# ---
